### PR TITLE
Fix use_cache.json with corrupted python causes `pdm use` error

### PR DESCRIPTION
**Pull request by linw1995**
## Pull Request Check List

- [x] A news fragment is added in `news/` describing what is new.
- [ ] Test cases added for changed code.

## Describe what you have changed in this PR.

Before

```plain
[⚡ PIPE] ❯ pdm use -f 3.10 -v
Traceback (most recent call last):
  File "/Users/linw1995/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/Users/linw1995/Documents/opensources/pdm/pdm/core.py", line 233, in main
    return Core().main(args)
  File "/Users/linw1995/Documents/opensources/pdm/pdm/core.py", line 168, in main
    raise cast(Exception, err).with_traceback(traceback)
  File "/Users/linw1995/Documents/opensources/pdm/pdm/core.py", line 163, in main
    f(options.project, options)
  File "/Users/linw1995/Documents/opensources/pdm/pdm/cli/commands/use.py", line 29, in handle
    actions.do_use(
  File "/Users/linw1995/Documents/opensources/pdm/pdm/cli/actions.py", line 571, in do_use
    cached_python = PythonInfo.from_path(use_cache.get(python))
  File "/Users/linw1995/Documents/opensources/pdm/pdm/models/python.py", line 27, in from_path
    raise InvalidPyVersion(f"Invalid Python interpreter: {path}")
pdm.exceptions.InvalidPyVersion: Invalid Python interpreter: /Users/linw1995/.pyenv/versions/3.10.0/bin/python3.10
```

After

image

### CODE REVIEW

**Comment by frostming** [comment]
Good catch, it might solve the random failure in recent CI ;)

**Comment by frostming** [comment]
Sorry, please rebase on the main branch

File: pdm/project/core.py [hunk 1]
```diff
@@ -581,6 +586,19 @@ def make_candidate_info_cache(self) -> CandidateInfoCache:
...
+        Helper method to not yield corrupted python info.
+        """
+        try:
+            yield PythonInfo.from_path(path)
+        except InvalidPyVersion:
+            self.core.ui.echo(
+                f"This interpreter {str(path)!r} is corrupted, skip.",
+                fg="yellow",
+                err=True,
+            )
```
Comment (frostming) [inline pdm/project/core.py:600]: Are we going to ignore all these errors? I am afraid we will fail to catch bugs and select the wrong Python interpreters in the end.

Comment (linw1995) [reply pdm/project/core.py:600]: It needs to refactor the class method `PythonInfo.from_path` and add an `invalid` attribute into `PythonInfo` class. So when a user selects a corrupted version, the `do_use` function can raise `InvalidPyVersion` error.

It sounds better way to resolve this problem.

**Pull request closed without merging**